Developer-Instructions.md

```diff
@@ -60,42 +56,30 @@ If you plan to enable the automated Personalize campaign creation process at dep
 }
 ```
 
-#### Staging for Deployment
+### Staging for Deployment
+
+We highly recommend to use a Python 3 virtual environment. Current supported version of python is 3.12 (other versions may work but we haven't tested all versions)
 
-It is advisable to use a Python 3 virtual environment to do this and the scripts assume that the executable pip is the Python 3 version of pip so if necessary you may need to install pip into that virtual environment (if your system defaults to a Python 2 version of pip).
+```bash
+python3.12 -m venv .venv/
+source .venv/bin/activate
+```
 
 The [stage.sh](stage.sh) script at the root of the repository must be used to upload the deployment resources to your staging S3 bucket if you use this option. The shell uses the local AWS credentials to build and push resources to your custom bucket. 
 
 Example on how to stage your project to a custom bucket and path (note the path is optional but, if specified, must end with '/'):
 
 ```bash
-./stage.sh MY_CUSTOM_BUCKET S3_PATH/
+./stage.sh MY_CUSTOM_BUCKET S3_PATH/ --private-s3 --skip-virtualenv
 ```
+NB:   --private-s3 is required to deploy to a private bucket (this is the recommended approach)
```

Comment:
Should we remove this option completely (i.e., always private)? What is the scenario where public is necessary?